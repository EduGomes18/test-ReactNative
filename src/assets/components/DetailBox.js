import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import { COLORS } from '../styles/colors';
import Icon from './Icon';

const Container = styled.View`
  height: 55px;
  margin: 8px 0px;
  background-color: ${COLORS.white};
  border-radius: 8px;
  border: 1px solid ${COLORS.blueLight2};
  flex-direction: row;
`;
const ContentLeft = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
`;
const ContentRight = styled.View`
  flex: 5;
  justify-content: center;
`;
const TitleRight = styled.Text`
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  line-height: 22px;
  color: ${COLORS.blueDark};
`;
const SubtitleRight = styled.Text`
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 18px;
  color: ${COLORS.blueLight4};
`;

const DetailBox = ({ title, subtitle }) => (
  <Container>
    <ContentLeft>
      <Icon name="calendarChecked" width={24} height={24} />
    </ContentLeft>
    <ContentRight>
      <TitleRight>{title}</TitleRight>
      <SubtitleRight>{subtitle}</SubtitleRight>
    </ContentRight>
  </Container>
);

DetailBox.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired
};

export default DetailBox;
