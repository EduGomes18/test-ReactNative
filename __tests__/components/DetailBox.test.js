import React from 'react';
import { shallow } from 'enzyme';
import DetailBox from '../../src/assets/components/DetailBox';

const defaultProps = {
  title: 'Testing title',
  subtitle: 'Testing subtitle'
};

test('DetailBox should render correctly', () => {
  const wrapper = shallow(<DetailBox {...defaultProps} />);

  expect(wrapper).toMatchSnapshot();
});
