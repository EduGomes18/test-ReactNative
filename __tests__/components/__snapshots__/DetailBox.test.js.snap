// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailBox should render correctly 1`] = `
<Styled(View)>
  <Styled(View)>
    <Icon
      height={24}
      name="calendarChecked"
      width={24}
    />
  </Styled(View)>
  <Styled(View)>
    <Styled(Text)>
      Testing title
    </Styled(Text)>
    <Styled(Text)>
      Testing subtitle
    </Styled(Text)>
  </Styled(View)>
</Styled(View)>
`;
